pkg/tests/apis/provisioning/repository_test.go

```diff
@@ -682,3 +683,86 @@ func TestIntegrationProvisioning_ImportAllPanelsFromLocalRepository(t *testing.T
 	require.Error(t, err, "should delete the internal resource")
 	require.True(t, apierrors.IsNotFound(err))
 }
+
+func TestIntegrationProvisioning_DeleteRepositoryAndReleaseResources(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	helper := runGrafana(t)
+	ctx := context.Background()
+
+	const repo = "gh-repo"
+	testRepo := TestRepo{
+		Name:               repo,
+		Template:           "testdata/github-readonly.json.tmpl",
+		Target:             "folder",
+		ExpectedDashboards: 3,
+		ExpectedFolders:    3,
+	}
+	helper.CreateRepo(t, testRepo)
+
+	// Checking resources are there and are managed
+	foundFolders, err := helper.Folders.Resource.List(ctx, metav1.ListOptions{})
+	require.NoError(t, err, "can list folders")
+	for _, v := range foundFolders.Items {
+		assert.Contains(t, v.GetAnnotations(), utils.AnnoKeyManagerKind)
+		assert.Contains(t, v.GetAnnotations(), utils.AnnoKeyManagerIdentity)
+	}
+
+	foundDashboards, err := helper.DashboardsV1.Resource.List(ctx, metav1.ListOptions{})
+	require.NoError(t, err, "can list dashboards")
+	for _, v := range foundDashboards.Items {
+		assert.Contains(t, v.GetAnnotations(), utils.AnnoKeyManagerKind)
+		assert.Contains(t, v.GetAnnotations(), utils.AnnoKeyManagerIdentity)
+		assert.Contains(t, v.GetAnnotations(), utils.AnnoKeySourcePath)
+		assert.Contains(t, v.GetAnnotations(), utils.AnnoKeySourceChecksum)
+	}
+
+	_, err = helper.Repositories.Resource.Patch(ctx, repo, types.JSONPatchType, []byte(`[
```

Comment:
@MissingRoberto AFAICS the flow for releasing resources in GitSync is:
- Update Repo (using a PATCH here, I can see the flow in FE does a PUT) , removing the `remove-orphan-resources` finaliser in favour of `release-orphan-resources` 
- Delete Repo
am I right?

Reply:
correct.